src/main/java/es/unizar/smartcampuz/application/controller/DashboardController.java

```diff
@@ -64,47 +68,52 @@
 
     @PostMapping("/reservation")
     @ResponseBody
-    public ResponseEntity<String> newReservation (HttpServletRequest request) throws IOException{
-        String location;
-        String email;
-        String description;
-        int day;
-        int month;
-        boolean [] requestedHours;
-
-        JSONObject json;
-        JSONArray jRequestedHours;
-
-        try{
-            json = JsonService.readJson(request.getReader());
-            location = json.getString("location");
-            email = json.getString("email");
-            description = json.getString("description");
-            day = json.getInt("day");
-            month = json.getInt(("month"));
-            jRequestedHours = json.getJSONArray("requestedHours");
-        }
-        catch (Exception e){
-            return new ResponseEntity<>("\"Error interno en el servidor.\"", HttpStatus.INTERNAL_SERVER_ERROR);
-        }
-        requestedHours = JsonService.JSONArrayToBooleanArray(jRequestedHours);
+    public ResponseEntity<String> newReservation (@RequestParam String location, @RequestParam String email,
+                                                  @RequestParam String description, @RequestParam int month, @RequestParam int day,
+                                                  @RequestParam boolean[] requestedHours) throws IOException, ParseException{
+
+//        JSONObject json;
```

Comment:
Yes. Actually, JSON accepts 1/0 and true/false notation so it doesn't mind.

Comment deleted.